include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -0,0 +1,539 @@
+                                           size_t                      pi_start,
+                                           size_t e12_value);
+
+  template <typename T, typename F, typename... Args>
+  void setup(T& tc, size_t num_gens, F func, Args... args) {
+    tc.set_number_of_generators(num_gens);
+    for (auto const& w : func(args...)) {
+      tc.add_pair(w.first, w.second);
+    }
+  }
+
+  template <typename T>
+  T make_group(T& M) {
+    std::string              id = "e";
+    std::unordered_set<char> self_inverse;
+    self_inverse.emplace('e');
+
+    for (auto it = M.cbegin_rules(); it != M.cend_rules(); ++it) {
+      if (it->second == id && it->first.size() == 2
+          && it->first[0] == it->first[1]) {
+        // Remove self-inverse elements
+        self_inverse.emplace(it->first[0]);
+      } else if (it->first == id && it->second.size() == 2
+                 && it->second[0] == it->second[1]) {
+        // Remove self-inverse elements
+        self_inverse.emplace(it->second[0]);
+      }
+    }
+
+    std::string alphabet;
+    std::string inverse;
+
+    for (auto const& letter : self_inverse) {
+      alphabet += letter;
+      inverse += letter;
+    }
+
+    for (auto const& letter : M.alphabet()) {
+      if (self_inverse.find(letter) == self_inverse.cend()) {
+        alphabet += letter;
+        alphabet += std::toupper(letter);
+        inverse += std::toupper(letter);
+        inverse += letter;
+      }
+    }
+
+    T G;
+    G.set_alphabet(alphabet);
+    G.set_identity(id);
+    G.set_inverses(inverse);
+    for (auto it = M.cbegin_rules(); it < M.cend_rules(); ++it) {
+      if ((it->second != id || it->first.size() != 2
+           || it->first[0] != it->first[1])
+          && (it->first != id || it->second.size() != 2
+              || it->second[0] != it->second[1])) {
+        G.add_rule(*it);
+      }
+    }
+    return G;
+  }
+
+  template <typename T>
+  T AlternatingGroupMoore(size_t n) {
+    static const std::string alphabet = "abcdfghijklmnopqrstuvwxyz";
+    T                        x;
+    std::string              id = "e";
+    x.set_alphabet(id
+                   + std::string(alphabet.cbegin(), alphabet.cbegin() + n - 2));
+
+    std::string const& a = x.alphabet();
+    x.add_rule(std::string({a[1]}) + a[1] + a[1], id);
+
+    for (size_t i = 2; i <= n - 2; ++i) {
+      x.add_rule(std::string({a[i - 1]}) + a[i] + a[i - 1] + a[i] + a[i - 1]
+                     + a[i],
+                 id);
+    }
+
+    for (size_t j = 1; j <= n - 4; ++j) {
+      x.add_rule(std::string({a[j]}) + a[j], id);
+    }
+
+    for (size_t k = 3; k <= n - 3; ++k) {
+      for (size_t j = 1; j < k - 1; ++j) {
+        x.add_rule(std::string({a[j]}) + a[k] + a[j] + a[k], id);
+      }
+    }
+    return make_group(x);
+  }
+
+  template <typename T,
+            typename
+            = std::enable_if_t<std::is_base_of<PresentationBase, T>::value>>
```

Comment:
```suggestion
            = std::enable_if_t<std::is_same<Presentation<word_type>, T>::value>>
```
As we discussed just now, this actually doesn't work for `std::string` so we should exclude that for now.